docs/user-guide/docker-compose.md

```diff
@@ -0,0 +1,380 @@
+
+```bash
+# start all new containers (eg. auth)
+docker-compose up -d
+
+# restart the api gateway to pick up new configuration settings
+docker-compose up -d -V ambassador
+
+# curl the quote-of-the-moment service without an auth header
+curl -I localhost:8080/qotm/quote/1
+
+# the response should look like this
+HTTP/1.1 401 Unauthorized
+x-powered-by: Express
+x-request-id: da059f3e-7b9e-4d98-8428-f4f8ca742af7
+www-authenticate: Basic realm="Ambassador Realm"
+content-type: text/html; charset=utf-8
+content-length: 0
+etag: W/"0-2jmj7l5rSw0yVb/vlWAYkK/YBwk"
+date: Fri, 17 Aug 2018 22:25:38 GMT
+x-envoy-upstream-service-time: 1
+server: envoy
+
+# now try with a specific username and password
+curl -I --user username:password localhost:8080/qotm/quote/1
+
+# the response should be a 200
+HTTP/1.1 200 OK
+content-type: application/json
+x-qotm-session: 5b75f31f-1155-4827-ab28-74d0f98573da
+content-length: 217
+server: envoy
+date: Fri, 17 Aug 2018 22:29:43 GMT
+x-envoy-upstream-service-time: 2
+```
+
+## 5. Tracing
+
+As a final example we will configure ambassador to send Zikin traces to Jaeger. Integrating Zipkin into your services can be a vital glimpse into the performance bottlenecks of a distributed system.
+
+### Add the Jaeger container to the docker-compose.yaml file
+
+Building on our original `docker-compose.yaml` file, we can add a new service called `tracing` to the list:
+
+```yaml
+version: '3.5'
+
+services:
+  ambassador:
+    image: quay.io/datawire/ambassador:0.38.0
+    ports:
+    # expose port 80
+    - 80
+    volumes:
+    # mount a volume where we can inject configuration files
+    - ./config:/ambassador/ambassador-config
+  qotm:
+    image: datawire/qotm:1.3
+    ports:
+    - 5000
+  auth:
+    image: datawire/ambassador-auth-service:latest
+    ports:
+    - 3000
+  tracing:
+    image: jaegertracing/all-in-one:latest
+    environment:
+      COLLECTOR_ZIPKIN_HTTP_PORT: 9411
+    ports: 
+      - 5775:5775/udp
+      - 6831:6831/udp
+      - 6832:6832/udp
+      - 5778:5778
+      - 16686:16686
+      - 14268:14268
+      - 9411:9411
+```
+
+### Create a tracing configuration file for Ambassador
+
+Add a new configuration file `config/tracing.yaml` with these contents:
+
+```yaml
+---
+apiVersion: ambassador/v0
+kind: TracingService
+name: tracing
+service: tracing:9411
+driver: zipkin
+```
+
+This will forward all of Ambassador's traces to the `tracing` service.
+
+### Make requests and observe the traces
+
+After reloading the docker containers and configuration we should be able to make requests to the qotm service and see the traces in the Jaeger front-end UI.
+
+```bash
+# start all new containers (eg. tracing)
+docker-compose up -d
+
+# restart the api gateway to pick up new configuration settings
+docker-compose up -d -V ambassador
+
+# curl the quote-of-the-moment service as many times as you would like
+curl -I localhost:8080/qotm/quote
```

Comment:
`curl localhost:8080/qotm/quote` is not accessible. Change to either `curl localhost:8080/qotm/quote/1` for select quote or `curl localhost:8080/qotm/` for random quote.